media.go

```diff
@@ -117,15 +116,14 @@ type producerConsumer[T any, U any] struct {
 // for error handling logic based on consecutively retrieved errors).
 type ErrorHandler func(ctx context.Context, mediaErr error)
 
-// PropertiesFromMediaSource attempts to return properties from the given MediaSource
-// if it has an underlying, online driver.
-func PropertiesFromMediaSource[T, U any](src MediaSource[T]) []prop.Media {
+// DriverFromMediaSource returns the underlying driver from the given media source.
+func DriverFromMediaSource[T, U any](src MediaSource[T]) (driver.Driver, error) {
```

Comment:
I'd prefer we don't expose a mutable river here. It sounds like you want Info so we can delete this current method and add an InfoFromMediaSource

Reply:
Done.